server/src/server.ts

```diff
@@ -140,12 +265,16 @@ connection.onDidChangeConfiguration((change) => {
 });
 
 function validateTextDocument(textDocument: TextDocument): void {
-    connection.console.log(`Yseop.vscode-yseopml − Parsing ${textDocument.uri}`);
     const textDocUri = textDocument.uri;
-    const diagnostics: Diagnostic[] = [];
+    const docContent = textDocument.getText();
+    parseFile(textDocUri, docContent);
+}
 
+function parseFile(textDocUri: string, docContent: string) {
+    connection.console.log(`Yseop.vscode-yseopml − Parsing ${textDocUri}`);
+    const diagnostics: Diagnostic[] = [];
     // Create the lexer and parser
-    const inputStream = CharStreams.fromString(textDocument.getText());
+    const inputStream = CharStreams.fromString(docContent, textDocUri);
     const lexer = new YmlLexer(inputStream);
     const tokenStream = new CommonTokenStream(lexer);
     const parser = new YmlParser(tokenStream);
```

Comment:
Do we need these for every file type?
Hum I suppose most of our files are YML anyway so the gain would be negligible.

Reply:
> Do we need these for every file type?

Unfortunately, yes. These objects are all required to perform the parsing. The parsing will be done only for certain types of files, so it should be OK to leave it that way. This is controlled by the extension itself.